src/ntp.rs

```diff
@@ -1,10 +1,29 @@
 use nom::{be_i8,be_u8,be_u16,be_u32,be_u64};
+use std::mem::transmute;
+
+#[derive(Debug,PartialEq,Eq)]
+pub enum NtpMode {
+    Reserved = 0,
+    SymmetricActive = 1,
+    SymmetricPassive = 2,
+    Client = 3,
+    Server = 4,
+    Broadcast = 5,
+    NtpControlMessage = 6,
+    Private = 7,
+}
+
+impl NtpMode {
+    pub fn new(m: u8) -> NtpMode {
+        unsafe { transmute(m) }
```

Comment:
This line seems dangerous - what happens if m is not in [0..7] ?
In my other parsers, I usually keep the field as u8, and declare the enum. Using enum_primitive, I had functions from_u8, that return an Option<T> instead of T, because the conversion could fail.
We can also add (impl) a function to get the value as NtpMode to simplify accessing value.